meetings/2017-08-10.md

```diff
@@ -0,0 +1,98 @@
+
+Can’t set an “auto-disclosure” after X days of filings in the settings (though
+you can after a bug was resolved), but probably possible with the API.
+Normally disclosure requires an intentional act within the report UI.
+
+Full API (https://api.hackerone.com).  Probably not a Node.js library for it
+yet (Ruby, Python, Go, etc. are present)
+
+Can issue CVEs for both node and npmjs.org packages if we choose to go down
+that route.
+
+Two possible use-cases:
+1. Node vulnerabilities
+2. Npmjs.org package vulnerabilities
+
+For npmjs.org packages, HackerOne looks to have a great feature set, CVE
+issuance and the rich permission model among them.
+
+
+I’d like to see some investigation of the API to ensure we can pull issues out
+as JSON for storage in github.com, but otherwise I think we should try it out.
+
+For node vulnerabilities, we’ll have to pitch HackerOne to the core members of
+the Node Security Response Team, Ben and Fyodor at least. The thing they might
+find most compelling for use with node core relates to permissions, since the
+Response team size has been described as too large recently. Use of HackerOne
+will allow:
+- Web or email submission of vulnerabilities, with the submitter being allowed
+  to view and discuss the submission directly using the web UI.
+- The conversation history to be made public when a vulnerability is
+  publicized: ATM all conversations about vulnerabilities are private for ever,
+  unnecessarily.
+- A triage team of 3 or 4 people who can see every submission, and either
+  reject it as not a vulnerability, or assign it to the correct people to work on
+  it further (those people would get access to only the one specific pre-release
+  vulnerabilty). This would allow, for example, HTTP experts who are not on the
+  Response team to be brought into the discussion of an HTTP issue. It would also
+  allow the setup of a team to further triage npmjs.org package vulnerability
+  reports, so the response team can re-assign non-node core vulnerabilities.
+
+## rubysec organization
+
+- Github: https://github.com/rubysec/ruby-advisory-db
+- Stores vuln per file, covers both gems (`gems/`) and ruby runtimes (`rubies`)
+- Every vuln has a CVE or a OSVDB, but the OSF (open security foundation) has
+  closed shop, and all the OSVDB links are now dead
+- Vuln files are named after the CVE/OSVDB ID, so they must request one for every
+  submitted vuln. @reedloden of HackerOne manages this process, allocating them
+  for ruby. It involves filling in a form of basic information, and he offered
+  to do this for Node.js as well.
+- Vuln schema: https://github.com/rubysec/ruby-advisory-db#schema
+- Vulns can be submitted via PR, or a form (https://rubysec.com/advisories/new),
+  but there isn’t any information on what the disclosure policy is. The form
+  suggests to me that they expect to see only reports for issues that are already
+  publically known and reported to the gem authors (and possibly already fixed).
```

Comment:
Yeah, RubySec mostly dealt with tracking existing vulnerabilities that were already fixed (or at least widely-known). Rarely did we ever act as a middle-man with disclosing issues to specific gems, mostly because we would just point people to http://guides.rubygems.org/security/#reporting-security-vulnerabilities (which, now that I read it, is outdated). However, in the Node.js world, I know that NSP has actively acted as that middle-man, which is fine, but it does require coordination work on somebody's part to reach out to the appropriate module author and get issues fixed.